x-pack/plugins/spaces/server/plugin.ts

```diff
@@ -126,13 +126,22 @@ export class SpacesPlugin
       initializerContext.config.create<ConfigType>(),
       this.onCloud$,
     ]).pipe(
-      map(
-        ([config, onCloud]): ConfigType => ({
+      map(([config, onCloud]): ConfigType => {
+        let allowSolutionVisibility = config.allowSolutionVisibility;
+
+        // We only allow "solution" to be set on cloud environments, not on prem
+        // unless the forceSolutionVisibility flag is set
+        if (config.forceSolutionVisibility) {
+          allowSolutionVisibility = true;
+        } else if (!onCloud) {
+          allowSolutionVisibility = false;
+        }
+
+        return {
           ...config,
-          // We only allow "solution" to be set on cloud environments, not on prem
-          allowSolutionVisibility: onCloud ? config.allowSolutionVisibility : false,
-        })
-      )
+          allowSolutionVisibility,
+        };
+      })
     );
```

Comment:
__optional nit:__ maybe it's just me, but the shorter version that's aligned with the comment, is "parsed" slightly easier when you read it:
```ts
    ]).pipe(
      map(
        ([config, onCloud]): ConfigType => ({
          ...config,
          // We only allow "solution" to be set on cloud environments, not on prem
          // unless the forceSolutionVisibility flag is set.
          allowSolutionVisibility:
            (onCloud && config.allowSolutionVisibility) || config.forceSolutionVisibility,
        })
      )
    );
```

Reply:
Good suggestion, addressed in https://github.com/elastic/kibana/pull/191743/commits/37fbf7b2e74f3f531a88ffbd54bbeecd4517b276